src/reactapp/src/components/AuthenticatedRoute.js

```diff
@@ -32,10 +32,17 @@ const AuthenticatedRoute = ({ element: authenticatedComponent, ...rest }) => {
         navigate("/login");
         sessionStorage.clear();
       }
-    }
-    );
+      if (roles && !roles.includes('*') && !isLoggedInUserInRole(roles)) {
+        setAlertMessageText("You do not have access to the requested page");
```

Comment:
Please replace with "You do not have access to the requested page, contact your administrator if you think this is in error."